File: easy/3701.py
"""3701. Compute Alternating Sum"""

from typing import List


class Solution:
    def alternatingSum(self, nums: List[int]) -> int:
        out: int = 0
        sign: int = 1

        for num in nums:
            out += sign * num
            sign *= -1

        return out

File: easy/3701.rs
// 3701. Compute Alternating Sum
// https://leetcode.com/problems/compute-alternating-sum/
// Weekly Contest 470

impl Solution {
    pub fn alternating_sum(nums: Vec<i32>) -> i32 {
        let mut out: i32 = 0;

        for idx in 0..nums.len() {
            if idx & 1 == 0 {
                out += nums[idx];
            } else {
                out -= nums[idx];
            }
        }

        out
    }
}
